### PR TITLE
Yan's challenge solution

### PR DESCRIPTION
**Pull request by yanekx**
## Description

I've done: 

- Make the participants controller index show only accepted ones. 
- Make the participants controller able to handle the creation of them.
- Full test coverage. 

I've wanted to add but it's out of the scope of this challenge: 
- Some auth. 
- Generalized way of serializing errors. 
- Design API using https://jsonapi.org/ specification and using (https://github.com/Netflix/fast_jsonapi) gem. 
-   some DB constraints and locks. 
- etc. 

Usually, I spent hours to just discuss the code design of the whole project, but at this challenge, I've been restricted by only 2 hrs., so I decided to make my design as simple as possible and also I saw that there're no DB constraints except `gender` one, so I decided that I don't have to manage other blank fields. There's a lot of things which I wanted to make better, but unfortunately, it's out of 2hrs challenge C: Thanks for reviewing this!

## How Has This Been Tested?

To test this you could run `bundle install`, `rake db:test:prepare` & `rspec` C:

### CODE REVIEW

File: app/controllers/api/v1/participants_controller.rb [hunk 1]
```diff
@@ -1,9 +1,43 @@
 module Api::V1
   class ParticipantsController < ApplicationController
-
     def index
-      render json: Participant.all
+      render json: Participant.accepted
```
Comment (ortegacmanuel) [inline app/controllers/api/v1/participants_controller.rb:4]: @yanekx nice scope!

File: app/controllers/api/v1/participants_controller.rb [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+    end
+
+    def create
+      create_service.call(participants_params)
```
Comment (ortegacmanuel) [inline app/controllers/api/v1/participants_controller.rb:8]: We love services! :+1: really clean approach!

File: app/models/participant.rb [hunk 3]
```diff
@@ -1,3 +1,11 @@
 class Participant < ApplicationRecord
   enum gender: { not_given: 0, male: 1, female: 2 }
+  enum status: { pending: 0, accepted: 1, rejected: 2 }
+
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, allow_blank: true
```
Comment (ortegacmanuel) [inline app/models/participant.rb:5]: Nice work adding validations!

File: app/services/participants/create_service.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
+class Participants::CreateService
```
Comment (ortegacmanuel) [inline app/services/participants/create_service.rb:1]: Nice work in both services!

**Comment by ortegacmanuel** [comment]
@yanekx great work! thanks for sending in a submission. We did a review but don't have much to say, your code is really good. Love the services, test coverage.. :+1:  

we'll get in contact soon. Thanks!

**Comment by yanekx** [comment]
> @yanekx great work! thanks for sending in a submission. We did a review but don't have much to say, your code is really good. Love the services, test coverage.. 👍
> 
> we'll get in contact soon. Thanks!
Thank you for the feedback @ortegacmanuel! I appreciate your quick response. I'd love to talk about the code at the meeting C:

**Pull request closed without merging**